client/src/components/logViewer/LogChannel.ts

```diff
@@ -0,0 +1,50 @@
+// Copyright © 2023, SAS Institute Inc., Cary, NC, USA.  All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { OutputChannel, l10n, window } from "vscode";
+
+import runStore, { subscribe } from "../../stores/run";
+import {
+  showLogOnExecutionFinish,
+  showLogOnExecutionStart,
+} from "../utils/settings";
+import { appendLog } from "./LogViewer";
+
+let outputChannel: OutputChannel;
+
+const { unsetProducedLogOutput, setProducedLogOutput } = runStore.getState();
+
+export const LogFn = (logs) => {
+  if (!outputChannel) {
+    outputChannel = window.createOutputChannel(l10n.t("SAS Log"), "sas-log");
+  }
+
+  for (const line of logs) {
+    appendLog(line.type);
+    outputChannel.appendLine(line.line);
+  }
+
+  setProducedLogOutput();
+};
+
+subscribe(
+  (state) => state.isRunning,
+  (running, prevRunning) => {
+    if (!running && prevRunning) {
+      if (showLogOnExecutionFinish()) {
+        outputChannel?.show(true);
+        unsetProducedLogOutput();
```

Comment:
This call should be independant of `if (showLogOnExecutionFinish())`, right?

Another thing is that, when running starts, that is
`if (running && !prevRunning)`
it should **also** `unsetProducedLogOutput()`
because, if user expanded the library view and produced log, there's no "done running" happened.

Reply:
Thanks @scnwwu. I've refactored the approach to split run and log into separate "slices" of the global store to clean things up a bit. I've also split the session log and code execution log action into separate zustand actions so that we can evaluate the "show log" options for code execution log since it seemed like we didnt want to be showing the output panel for session log outputs. See https://github.com/sassoftware/vscode-sas-extension/pull/713/commits/89aca224ac760fe0ed04cc79d8baf8c980113f6d. Still working on writing tests for the actions and initial state pieces.